### PR TITLE
init/on_init fixes

### PR DESCRIPTION
**Pull request by shreyaxh**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Init.hs [hunk 1]
```diff
@@ -22,7 +22,7 @@ castVehicleVariant mbVehCategory mbVehVariant = case (mbVehCategory, mbVehVarian
   _ -> Nothing
 
 castPaymentType :: MonadFlow m => Text -> m DMPM.PaymentType
-castPaymentType "ON_FULFILLMENT" = return DMPM.ON_FULFILLMENT
+castPaymentType "ON-FULFILLMENT" = return DMPM.ON_FULFILLMENT
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Init.hs:25]: can we replace `castPaymentType` with `decodeFromText`. We need to have FromJSON instance for this to work

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnInit.hs [hunk 2]
```diff
@@ -187,12 +187,12 @@ tfItems res =
...
-tfItemDescriptor :: DInit.InitRes -> Maybe Spec.Descriptor
-tfItemDescriptor res =
+tfItemDescriptor :: Maybe Spec.Descriptor
+tfItemDescriptor =
   Just
     Spec.Descriptor
-      { descriptorCode = Nothing,
-        descriptorShortDesc = Just $ Common.mkItemId res.transporter.shortId.getShortId res.booking.vehicleVariant,
+      { descriptorCode = Just "RIDE", -- TODO : maybe make Enum for it?
+        descriptorShortDesc = Nothing,
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnInit.hs:195]: Keep the short description

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Init.hs [hunk 3]
```diff
@@ -21,9 +21,9 @@ castVehicleVariant mbVehCategory mbVehVariant = case (mbVehCategory, mbVehVarian
   (Just "CAB", Just "TAXI_PLUS") -> Just VehVar.TAXI_PLUS
   _ -> Nothing
 
-castPaymentType :: MonadFlow m => Text -> m DMPM.PaymentType
-castPaymentType "ON_FULFILLMENT" = return DMPM.ON_FULFILLMENT
-castPaymentType _ = throwM $ InvalidRequest "Unknown Payment Type"
+-- castPaymentType :: MonadFlow m => Text -> m DMPM.PaymentType
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/OnDemand/Utils/Init.hs:24]: Remove commented code

File: Backend/lib/beckn-spec/src/BecknV2/OnDemand/Enums.hs [hunk 4]
```diff
@@ -64,7 +65,11 @@ data PaymentStatus
   = -- ..fulfillments.payment.status
     PAID
   | NOT_PAID
-  deriving (Show, Eq, Generic, ToJSON, FromJSON)
+  deriving (Eq, Generic, ToJSON, FromJSON)
+
+instance Show PaymentStatus where
```
Comment (NikithShetty) [inline Backend/lib/beckn-spec/src/BecknV2/OnDemand/Enums.hs:70]: Add json parsing instances here as well

File: Backend/lib/beckn-spec/src/BecknV2/OnDemand/Enums.hs [hunk 5]
```diff
@@ -86,7 +116,29 @@ data OrderStatus
   | ACTIVE
   | COMPLETE
   | CANCELLED
-  deriving (Show, Eq, Generic, ToJSON, FromJSON, Read)
+  deriving (Eq, Generic, Read)
+
+instance Show OrderStatus where
+  show SOFT_CANCEL = "SOFT-CANCEL"
```
Comment (NikithShetty) [inline Backend/lib/beckn-spec/src/BecknV2/OnDemand/Enums.hs:122]: Use existing function `replaceUnderscores` instead of manual case match

**Review by NikithShetty — CHANGES_REQUESTED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs [hunk 6]
```diff
@@ -209,3 +214,15 @@ tfCustomer res =
                 personTags = Nothing
               }
       }
+
+tfCancellationTerms :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => DConfirm.DConfirmResp -> m [Spec.CancellationTerm]
+tfCancellationTerms res = do
+  let vehicleCategory = Utils.mapVariantToVehicle res.vehicleVariant
+  becknConfig <- QBC.findByMerchantIdDomainAndVehicle (Just res.transporter.id) (show Context.MOBILITY) vehicleCategory >>= fromMaybeM (InternalError "Beckn Config not found")
```
Comment (NikithShetty) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:221]: Keep transformers functions pure. Pass the relevant data to the transformers

Comment (shreyaxh) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnConfirm.hs:221]: Fixed this.

**Review by NikithShetty — APPROVED** [review]

**Pull request merged**